docs/protocol/specifications/blockchain.md

```diff
@@ -0,0 +1,613 @@
+  * [Merkle Root](#merkle-root)
+  * [Merkle Binary Tree](#merkle-binary-tree)
+  * [Merkle Patricia Tree](#merkle-patricia-tree)
+  * [Transactions Merkle Root](#transactions-merkle-root)
+  * [Assets Merkle Root](#assets-merkle-root)
+* [Entries](#entries)
+  * [Entry](#entry)
+  * [Entry ID](#entry-id)
+  * [Block Header](#block-header)
+  * [Block ID](#block-id)
+  * [Transaction Header](#transaction-header)
+  * [Transaction ID](#transaction-id)
+  * [Output 1](#output-1)
+  * [Spend 1](#spend-1)
+  * [Issuance 1](#issuance-1)
+  * [Mux 1](#mux-1)
+  * [Nonce](#nonce)
+  * [Time Range](#time-range)
+
+
+## Introduction
+
+This is a specification of the semantic data structures used by blocks and transactions. These data structures and rules are used for validation and hashing. This format is independent from the format for transaction wire serialization.
+
+### Block
+
+A **block** is a [block header](#block-header) together with a list of [transactions](#transaction).
+
+### Transaction
+
+A **transaction** is composed of a set of [entries](#entries). Each transaction must include one and only one [transaction header](#transaction-header), which references other entries in the transaction, which in turn can reference additional entries. 
+
+Every entry is identified by its [Entry ID](#entry-id). No two entries in a transaction may have the same [Entry ID].
+
+
+## Types
+
+### LEB128
+
+[Little Endian Base 128](https://developers.google.com/protocol-buffers/docs/encoding) encoding for unsigned integers typically used to specify length prefixes for arrays and strings. Values in range [0, 127] are encoded in one byte. Larger values use two or more bytes.
+
+### Integer
+
+A [LEB128](#leb128) integer with a maximum allowed value of 0x7fffffffffffffff (2<sup>63</sup> – 1) and a minimum of 0. A varint63 fits into a signed 64-bit integer.
+
+### String
+
+A binary string with a [LEB128](#leb128) prefix specifying its length in bytes.
+The maximum allowed length of the underlying string is 0x7fffffff (2<sup>31</sup> – 1).
+
+The empty string is encoded as a single byte 0x00, a one-byte string is encoded with two bytes 0x01 0xNN, a two-byte string is 0x02 0xNN 0xMM, etc. 
+
+### SHA3
+
+*SHA3* refers to the SHA3-256 function as defined in [FIPS202](https://dx.doi.org/10.6028/NIST.FIPS.202) with a fixed-length 32-byte output.
+
+This hash function is used throughout all data structures and algorithms in this spec,
+with the exception of SHA-512 (see [FIPS180](http://csrc.nist.gov/publications/fips/fips180-2/fips180-2withchangenotice.pdf)) used internally as function H inside Ed25519 (see [CFRG1](https://tools.ietf.org/html/draft-irtf-cfrg-eddsa-05)).
+
+### Hash
+
+A fixed-length 32-byte string.
+
+### List
+
+A `List` is encoded as a [Varstring31](#string) containing the serialized items, one by one, as defined by the schema. 
+
+Note: since the `List` is encoded as a variable-length string, its length prefix indicates not the number of _items_,
```

Comment:
Nice catch, but why "alas"? Unless there's some problem I'm not seeing with what we've implemented, it seems better. Finding the length in bytes would take extra work.

It might help if we avoid thinking of this whole thing as a serialization format; that's not what it is. (In hindsight, it would have made more sense to compute the txid as the hash of the entry hashes rather than the entries themselves. That'd be even less serial than what we have now.)